### PR TITLE
Added Azure module for a private endpoint

### PR DESCRIPTION
**Pull request by pipalmic**

### CODE REVIEW

**Review by patrik-pa4k — APPROVED** [review]

File: modules/azure/private_endpoint/main.tf [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+resource "azurerm_private_endpoint" "private_endpoint" {
+  name                = var.name
+  location            = var.location
+  resource_group_name = var.resource_group_name
+  subnet_id           = var.subnet_id
+
+  private_service_connection {
+    name                           = var.private_connection_name
+    private_connection_resource_id = var.private_connection_resource_id
+    subresource_names              = [var.private_connection_resource_subresource_type]
```
Comment (tom-reinders) [inline modules/azure/private_endpoint/main.tf:27]:
```suggestion
    subresource_names              = var.private_connection_subresource_names
```

File: modules/azure/private_endpoint/variables.tf [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+variable "private_connection_resource_id" {
+  type        = string
+  description = "The ID of the resource where the connection should be made to."
+}
+
+variable "private_connection_resource_subresource_type" {
+  type        = string
+  description = "The subresource type of the resource where the connection should be made to."
+}
```
Comment (tom-reinders) [inline modules/azure/private_endpoint/variables.tf:24]:
```suggestion
variable "private_connection_subresource_names" {
  type        = set(string)
  description = "The subresource names of the resource where the connection should be made to."
  default     = null
}
```

File: modules/azure/private_endpoint/main.tf [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    is_manual_connection           = false
+  }
+
+  private_dns_zone_group {
+    name                 = var.private_dns_zone_group_name
+    private_dns_zone_ids = [var.private_dns_zone_id]
```
Comment (tom-reinders) [inline modules/azure/private_endpoint/main.tf:33]:
```suggestion
    private_dns_zone_ids = var.private_dns_zone_ids
```

File: modules/azure/private_endpoint/variables.tf [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+variable "private_connection_name" {
+  type        = string
+  description = "The name of the private connection name"
+}
+
+variable "private_dns_zone_id" {
+  type        = string
+  description = "The ID of the private DNS zone."
+}
```
Comment (tom-reinders) [inline modules/azure/private_endpoint/variables.tf:39]:
```suggestion
variable "private_dns_zone_ids" {
  type        = set(string)
  description = "The IDs of the private DNS zones."
}
```

**Review by tom-reinders — CHANGES_REQUESTED** [review]
A few improvements, the rest looks great!

**Comment by pipalmic** [comment]
Good points, that will make it even a bit more generic, updated.

**Review by tom-reinders — APPROVED** [review]

**Pull request merged**